plugins/modules/vmware_guest.py

```diff
@@ -1855,7 +1855,9 @@ def configure_network(self, vm_obj):
                 nic_change_detected = True
 
             net_obj = self.cache.get_network(network_name)
-            if hasattr(net_obj, 'portKeys'):
+            if hasattr(net_obj, 'portKeys') and \
```

Comment:
Please create a [changelog fragment](https://docs.ansible.com/ansible/latest/community/development_process.html#creating-a-changelog-fragment). See [this fragment](https://github.com/ansible/ansible/blob/c2f528b7686c9573aa8493357219e0bbe4416dd3/changelogs/fragments/reboot-change-default-boot-command.yaml) as an example.

Reply:
Also, add test cases for this change.

Reply:
> Also, add test cases for this change.

@Akasurde I don't think we have a way to run tests against VMWare Cloud on AWS in the CI at the moment. It looks a bit like we could substitute this with having NSX-T in the CI, but we don't have this, either.

@goneri Thoughts?

Reply:
VMC and NSX-T are not available. I meant we need to make sure that the already working code is not impacted by this change.

Reply:
@gruidl Yes and no.

Yes: I guess `vmware_guest_network` will run into more or less the same issue in VMware Cloud on AWS.

No: I don't think this should be just copy&paste'd over to `vmware_guest_network`. It would be better to refactor both `vmware_guest` and `vmware_guest_network` and move the networking stuff to a shared library like we did for storage devices in #909. DRY!

Unfortunately, that's harder to do and probably a lot more work :-/

Reply:
@skain80 I'm on VMC on Dell, cant find the version (yet).  However - all the new properties are there per the vmware KB.  As i said, your code here allowed for the VM to be deployed rather than erroring out - but its on a standard network of the same name.

Reply:
@skain80 @mariolenz UPDATE: The current code works as it is on Dell VMC v1.16.6.  The reason we were encountering the error condition was completely due to permissions.  If you arent calling the module with an account in the VMC's Cloudadmingroup (NOT just the cloudadmin role) - certain attributes were hidden, thus causing the error.